### ISSUE TITLE
ISR is not working when using Styled-Components

### ISSUE THREAD

**Issue by codekcv** [vercel/next.js#21759]
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

14.15.4

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**How are you deploying your application?**

Vercel

**Describe the Bug**

ISR will revalidate only once with styled-components. After that and then updating backend content, revalidating doesn't trigger anymore forever.

**Expected Behavior**

Should revalidate.

**To Reproduce**

1. Create a fresh create-next-app.
2. Install styled-components following this https://github.com/vercel/next.js/tree/canary/examples/with-styled-components
(with _documents, .babelrc, and babel dependency)
3. Create a Layout styled component to wrap the Component prop in _app.js
4. In index page getStaticProps, fetch for dynamic content.
5. Add revalidate: 1 key/value in returned object in getStaticProps.
`return {
props: {...yourFetchedData},
revalidate: 1
}`
6. Deploy at vercel.
7. Make changes in your content.
8. Re-visit page for at least 1s to trigger revalidation.
9. Wait for it to update...
10. it doesn't happen, you wait forever and nothing. (It might happen ONCE only, after that, start again from step 6 and no more...)

Thanks, please fix. This is a very big issue.

In _app.jsx

const Container = styled.div`
  background: red;
`;

const Layout = ({ children }) => <Container>{children}</Container>;

const MyApp = ({ Component, pageProps }) => (
  <Layout>
    <Component {...pageProps} />
  </Layout>
);

export default MyApp;

Extra note:
The dynamic content I'm testing is from updating blog using CMS(strapi).
At local production build using next build && next start... it does revalidate and update. But when on vercel it doesn't.

**Comment by codekcv** [comment]
not really.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
